knowledge-base/create-tables-programmatically.md

```diff
@@ -0,0 +1,161 @@
+---
+title: How to Create Tables Programmatically in Telerik Reporting
+description: Learn how to create tables from scratch at runtime using Telerik Reporting.
+type: how-to
+page_title: How to Create Tables Programmatically in Telerik Reporting
+meta_title: How to Create Tables Programmatically in Telerik Reporting
+slug: generating-runtime-tables-telerik-reporting
+tags: telerik reporting, runtime report creation, programmatic report, report generation, dynamic table
+res_type: kb
+---
+
+## Environment
+
+<table>
+    <tbody>
+        <tr>
+            <td> Product </td>
+            <td> Progress® Telerik® Reporting </td>
+        </tr>
+        <tr>
+            <td> Version </td>
+            <td> 19.1.25.521 </td>
+        </tr>
+    </tbody>
+</table>
+
+## Description
+
+This article demonstrates how to create tables programmatically in Telerik Reporting. While this approach offers full control over table structure and data binding, it is generally recommended to start with a designer-generated table and modify it programmatically if needed. The Telerik Report Designer handles complex layout and grouping logic, making it easier to maintain and extend reports.
+
+## Prerequisites
+
+Before creating a table programmatically, ensure you have a valid `Telerik.Reporting.Report` object with a predefined structure and a data source ready be set to the table. If not, follow the steps below to create them programmatically.
+
+### 1. Initializing the Report
+
+Create a new report object and configure its basic settings:
+
+````csharp
+var report = new Telerik.Reporting.Report();
+report.Name = "Report1";
+report.PageSettings.Margins = new Telerik.Reporting.Drawing.MarginsU(Telerik.Reporting.Drawing.Unit.Mm(20D), Telerik.Reporting.Drawing.Unit.Mm(20D), Telerik.Reporting.Drawing.Unit.Mm(20D), Telerik.Reporting.Drawing.Unit.Mm(20D));
+report.PageSettings.PaperKind = System.Drawing.Printing.PaperKind.A4;
+report.Width = Telerik.Reporting.Drawing.Unit.Cm(17D);
+````
+
+### 2. Adding Report Sections
+
+Most reports include three main sections:
+
+- **PageHeaderSection**: Appears at the top of each page.
```

Comment:
- When making a reference to an element of the code, use `monospace`. For example, the `PageHeaderSection` class.
- This comment applies to all three sections listed here.
```suggestion
- `PageHeaderSection`: Appears at the top of each page.
```